core/runtime/register_trt_op.cpp

```diff
@@ -10,80 +10,87 @@ namespace trtorch {
 namespace core {
 namespace runtime {
 
-// SM Compute capability <Compute Capability, Device Name> map
-const std::unordered_map<std::string, std::string>& get_dla_supported_SM() {
-  // Xavier SM Compute Capability
-  static std::unordered_map<std::string, std::string> dla_supported_SM = {{"7.2", "Xavier"}};
-  return dla_supported_SM;
-}
-
 // Checks if the context switch requred for device ID
 bool is_switch_required(const CudaDevice& curr_device, const CudaDevice& conf_device) {
   // If SM capability is not the same as configured then switch
   if ((curr_device.major != conf_device.major) || (curr_device.minor != conf_device.minor)) {
-    LOG_WARNING("Configured SM capability does not match with current device ID. Switching context");
+    LOG_WARNING(
+        "Configured SM capability " << conf_device.getSMCapability()
+                                    << " does not match with current device SM capability "
+                                    << curr_device.getSMCapability() << " (" << curr_device
+                                    << "). Switching device context");
     return true;
   }
 
   // GPU case
   if (conf_device.device_type == nvinfer1::DeviceType::kGPU) {
     if (curr_device.device_name != conf_device.device_name) {
       LOG_WARNING(
-          "TRTEngine compiled for " << conf_device.device_name << " but current CUDA device is "
-                                    << curr_device.device_name << ". Switching the device context");
+          "Program compiled for " << conf_device.device_name << " but current CUDA device is " << curr_device
+                                  << ". Attempting to switch device context for better compatibility");
```

Comment:
Here "better compatibility" shows that cards with different compute capabilities are compatible which is not the case.

How about "to a matching compute capablity"

Reply:
This is a device name switch, so like switching from RTX 3070 to RTX 3080, the compute capability should be the same